routes/api/contacts.js

```diff
@@ -46,26 +41,41 @@ router.delete("/:id", async (req, res, next) => {
     return res.status(404).json({ "message": "Not found"});
   }
   try {
-    removeContact(id);
+    await removeContact(id);
     return res.status(200).json({"message": "contact deleted"});
   } catch {
     return res.status(500).json({ "message": "Something went wrong" });
   }
 })
 
-router.put('/:id', async (req, res, next) => {
+router.put("/:id", async (req, res, next) => {
   const { id } = req.params;
- if (!id) {
-    return res.status(404).json({ "message": "Not found"});
+  
+  if (!id) {
+    return res.status(404).json({ "message": "Not found" });
+  }
+try {
+    const contactUpdate = await updateContact( id, req.body);
+    return res.status(200).json(contactUpdate);
+  } catch (error) {
+    next(error);
+    return res.status(500).json({ "message": "Something went wrong" });
+  }
+});
+
+router.patch("/:id/favorite", async (req, res, next) => {
+  const { id } = req.params;
+  if (!id) {
+    return res.status(404).json({ "message": "Not found" });
   }
-    const { error } = contactSchema.validate(req.body);
-    if (error) {
-      return res.status(400).json({"message": "missing fields"});
-    }
   try {
-    const contact = await updateContact(id, req.body);
-    return res.status(200).json(contact);
-  } catch {
+    if (typeof req.body.favorite !== "boolean") {
+      return res.status(400).send({ "message": "missing field favorite" });
+    }
+    const favoritetUpdate = await updateContact( id, req.body);
```

Comment:
literówka -> `favoriteUpdate`

Reply:
Dodatkową rzeczą jest to, że możliwe jest dodanie jakichkolwiek danych, i kontakt zostanie zaktualizowany za pomocą tych danych. Jedyną możliwością powinno być zmienienie pola `favorite`, z wyłączeniem innych pól